components/mediation/data-publishers/org.wso2.micro.integrator.observability/src/main/java/org/wso2/micro/integrator/observability/handler/metrichandler/MetricHandler.java

```diff
@@ -0,0 +1,388 @@
+
+    @Override
+    public boolean handleServerInit() {
+        getMetricReporter();
+        metricReporterInstance.initMetrics();
+
+        metricReporterInstance.serverUp(HOST, PORT, JAVA_HOME, JAVA_VERSION);
+        return true;
+    }
+
+    /**
+     * Load the MetricReporter class from the deployment.toml file if a user has defined a MetricReporter.
+     * Use default PrometheusReporter if the user hasn't defined a MetricReporter or an error occurs
+     * during custom MetricReporter class invocation.
+     */
+    private MetricReporter getMetricReporter() {
+        Map<String, Object> configs = ConfigParser.getParsedConfigs();
+        Object metricReporterClass = configs.get(MetricConstants.METRIC_HANDLER + "." + METRIC_REPORTER);
+        Class loadedMetricClass;
+
+        if (metricReporterClass != null) {
+            try {
+                loadedMetricClass = Class.forName(metricReporterClass.toString());
+                metricReporterInstance = (MetricReporter) loadedMetricClass.newInstance();
+                log.debug("The class " + metricReporterClass + "loaded successfully");
+            } catch (IllegalAccessException | ClassNotFoundException | InstantiationException e) {
+                log.error("Error in loading the class" + metricReporterClass.toString() +
+                        "Hence loading the default PrometheusReporter class ");
+                metricReporterInstance = loadDefaultPrometheusReporter();
+            }
+        } else {
+           metricReporterInstance = loadDefaultPrometheusReporter();
+        }
+        return metricReporterInstance;
+    }
+
+    /**
+     * Load the PrometheusReporter class by default.
+     */
+    private MetricReporter loadDefaultPrometheusReporter() {
+        metricReporterInstance = new PrometheusReporter();
+        log.debug("The class org.wso2.micro.integrator.obsrvability.handler.metrics.publisher.prometheus." +
+                "reporter.PrometheusReporter loaded successfully");
+
+        return metricReporterInstance;
+    }
+
+    @Override
+    public boolean handleRequestInFlow(MessageContext synCtx) {
+        org.apache.axis2.context.MessageContext axis2MessageContext = ((Axis2MessageContext) synCtx)
+                .getAxis2MessageContext();
+
+        if (null != synCtx.getProperty(SynapseConstants.PROXY_SERVICE)) {
+            String proxyName = axis2MessageContext.getAxisService().getName();
+
+            incrementProxyCount(proxyName);
+            startTimers(synCtx, proxyName, SynapseConstants.PROXY_SERVICE_TYPE, null);
+        } else if (null != synCtx.getProperty(SynapseConstants.IS_INBOUND)) {
+            String inboundEndpointName = synCtx.getProperty(SynapseConstants.INBOUND_ENDPOINT_NAME).toString();
+
+            incrementInboundEndPointCount(inboundEndpointName);
+            startTimers(synCtx, inboundEndpointName, MetricConstants.INBOUND_ENDPOINT,
+                    null);
```

Comment:
We can have this in the same line right? It does not exceed the line limit.

Reply:
Fixed